python/paddle/fluid/incubate/fleet/base/role_maker.py

```diff
@@ -281,25 +293,30 @@ def _server_num(self):
         return the current number of server
         """
         if self._check_role_generation():
-            if self.is_server():
-                return self._get_size() / 2
-        return 0
+            return self._get_size() / 2
```

Comment:
```
if not self._check_role_generation():
    self.generate_role()
return self._get_size() / 2
```
maybe more clearly?

Reply:
if role is not generated, just generate role. otherwise return _get_size() / 2. in MPISymetricRoleMaker, number of total process is worker_num + server_num, so _get_size() / 2 gets server number.